spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb

```diff
@@ -14,92 +14,129 @@
 
     credentials = {:userid => username, :password => password}
     @ems.authentications << FactoryBot.create(:authentication, credentials)
+    @inventory = nil
+  end
+
+  def reset_cache
+    @ems.reset_openstack_handle
+
+    require "fog/openstack"
+    Fog::OpenStack.instance_variable_set(:@version, nil)
+  end
+
+  def refresh_inventory
+    with_vcr do
+      reset_cache
+
+      @ems.refresh
+      @ems.network_manager.refresh
+      @ems.cinder_manager.refresh
+      @ems.swift_manager.refresh
+    end
+
+    @inventory ? assert_inventory_not_changed { @inventory } : @inventory = serialize_inventory
+    @ems.reload
+
+    assert_specific_vm
+    assert_specific_auth_key_pair
+    assert_specific_cloud_volume
+    assert_specific_host_aggregates
+    assert_specific_cloud_network
+    assert_specific_cloud_subnet
+    assert_specific_security_group
+    assert_specific_network_router
   end
 
   it ".ems_type" do
     expect(described_class.ems_type).to eq(:openstack)
   end
 
   it "will perform a full refresh" do
-    inventory = nil
     2.times do # Run twice to verify that a second run with existing data does not change anything
-      with_vcr do
-        @ems.reset_openstack_handle
-
-        require "fog/openstack"
-        Fog::OpenStack.instance_variable_set(:@version, nil)
+      refresh_inventory
+    end
+  end
 
-        @ems.refresh
-        @ems.network_manager.refresh
-        @ems.cinder_manager.refresh
-        @ems.swift_manager.refresh
+  context "when using an admin account for fast refresh" do
+    it "will perform a fast full refresh" do
+      ::Settings.ems_refresh.openstack.is_admin = true
+      ::Settings.ems_refresh.openstack_network.is_admin = true
```

Comment:
I know this is how the current test is written but it is much better to use stub_settings_merge:
```ruby
```suggestion
  context "when using an admin account for fast refresh" do
    before do
      stub_settings_merge(
        :ems_refresh => {
          :openstack => {:is_admin => true},
          :openstack_network => {:is_admin => true}
        }
      )
    end
    
    it "will perform a fast full refresh" do
```

And then you don't have to set it back down [[here]](https://github.com/ManageIQ/manageiq-providers-openstack/pull/819/files#diff-47be47aae1eb015036827f5434eed1ba3b3981cc68ad9e3967cf01311cfbfd37R67-R68)

Reply:
The reason being messing with the global state is a recipe for sporadic test failures